vllm/model_executor/models/internvl.py

```diff
@@ -253,12 +259,15 @@ def input_processor_for_internvl(ctx: InputContext, llm_inputs: LLMInputs):
                      multi_modal_data=multi_modal_data)
 
 
-def input_mapper_for_internvl(ctx: InputContext, data: object):
+def input_mapper_for_internvl(ctx: InputContext,
+                              data: object,
+                              *,
+                              max_dynamic_patch: Optional[int] = None):
     hf_config = ctx.get_hf_config()
 
     use_thumbnail = hf_config.use_thumbnail
     min_num = hf_config.min_dynamic_patch
-    max_num = hf_config.max_dynamic_patch
+    max_num = max_dynamic_patch or hf_config.max_dynamic_patch
```

Comment:
Maybe instead set it up in input processor/mapper, and require it to be an integer in `get_max_internvl_image_tokens`?